### PR TITLE
CFD-363 : events global overview

### PR DESCRIPTION
**Pull request by MPParsley**
global overview for events (landing page, future & past).
filter on interests & groups
sort future & past events on date & relevance

### CODE REVIEW

File: capacity4more/modules/c4m/field/c4m_field_og/includes/c4m_field_og.theme.node.inc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * Custom Display Suite function fields.
+ */
+
+/**
+ * DS function field.
+ *
+ * Renders a link to the group the current group node belongs to.
+ *
+ * @param $field
```
Comment (zero2one) [inline capacity4more/modules/c4m/field/c4m_field_og/includes/c4m_field_og.theme.node.inc]: Missing type + description.

File: capacity4more/modules/c4m/field/c4m_field_og/includes/c4m_field_og.theme.node.inc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @return string
```
Comment (zero2one) [inline capacity4more/modules/c4m/field/c4m_field_og/includes/c4m_field_og.theme.node.inc]: Missing description

File: capacity4more/modules/c4m/field/c4m_field_og/includes/c4m_field_og.theme.node.inc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  // Don't show this field in a group context.
+  $context = og_context();
+  if ($context) {
+    return;
+  }
+
+  // Only show this field for group nodes.
+  $entity = $field['entity'];
+  if (!og_is_group_content_type('node', $entity->type)) {
+    dsm($entity->type);
```
Comment (zero2one) [inline capacity4more/modules/c4m/field/c4m_field_og/includes/c4m_field_og.theme.node.inc]: Remove debug function.

**Comment by zero2one** [comment]
Build is failing due to [code style issues](https://travis-ci.org/capacity4dev/capacity4more/jobs/55187572#L211).

The group name is still missing in the list overview.

c4m-missing-group-name

**Comment by MPParsley** [comment]
global_events_past
global_events_upcoming

**Comment by MPParsley** [comment]
removed debug, fixed dochead and field settings, added screenshots

**Comment by zero2one** [comment]
[FILE: ...ules/c4m/field/c4m_field_og/includes/c4m_field_og.theme.node.inc](https://travis-ci.org/capacity4dev/capacity4more/jobs/55215936#L211)
```
----------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 4 LINES
----------------------------------------------------------------------
  5 | ERROR | [x] Doc comment short description must be on the first
    |       |     line
 15 | ERROR | [x] Separate the @param and @return sections by a blank
    |       |     line.
 15 | ERROR | [ ] Return comment must be on the next line
 17 | ERROR | [ ] Type hint "array" missing for $field
 19 | ERROR | [ ] Function return type is not void, but function is
    |       |     returning void here
```

**Pull request merged**